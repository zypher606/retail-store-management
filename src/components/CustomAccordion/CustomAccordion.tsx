import React, { forwardRef, useRef, useState } from 'react';
import { withStyles } from '@material-ui/core/styles';
import MuiAccordion from '@material-ui/core/Accordion';
import MuiAccordionSummary from '@material-ui/core/AccordionSummary';
import MuiAccordionDetails from '@material-ui/core/AccordionDetails';
import Typography from '@material-ui/core/Typography';
import Avatar from '@material-ui/core/Avatar';
import { Divider, RootRef } from '@material-ui/core';
import { fade, makeStyles } from '@material-ui/core/styles';

export const useStyles = makeStyles((theme: any) => ({
  detailExpanded: {
    maxHeight: '200px',
    transition: 'max-height 0.25s ease-in',
    overflow: 'hidden',
  },
  detailCollapsed: {
    maxHeight: '0px',
    transition: 'max-height 0.15s ease-out',
    overflow: 'hidden',
  }
}));
const Accordion = withStyles({
  root: {
    border: '1px solid rgba(0, 0, 0, .125)',
    boxShadow: 'none',
    '&:not(:last-child)': {
      borderBottom: 0,
    },
    '&:before': {
      display: 'none',
    },
    '&$expanded': {
      margin: 'auto',
    },
  },
  expanded: {},
})(MuiAccordion);

const AccordionSummary = withStyles({
  root: {
    backgroundColor: '#fff',
    // borderBottom: '1px solid rgba(0, 0, 0, .125)',
    marginBottom: -1,
    minHeight: 56,
    '&$expanded': {
      minHeight: 56,
    },
  },
  content: {
    '&$expanded': {
      margin: '12px 0',
    },
  },
  expanded: {},
})(MuiAccordionSummary);

const AccordionDetails = withStyles((theme) => ({
  root: {
    padding: theme.spacing(2),
    paddingLeft: theme.spacing(8),
  },
}))(MuiAccordionDetails);

interface ICustomAccordion {
  openDefault: boolean;
  fixed?: boolean;
  heading: any;
  body: any;
}

export const CustomAccordion = ({
  openDefault = false,
  fixed = false,
  heading,
  body,
  
}: ICustomAccordion) => {

  const [expanded, setExpanded] = useState<boolean>(openDefault);
  const classes = useStyles();

  const toggleExpanded = () => {
    if (fixed) return;
    setExpanded(!expanded);
  }

  return (
    <Accordion square expanded={true}>
      <AccordionSummary onClick={toggleExpanded} className={expanded ? 'is-expanded' : 'is-not-expanded'}>
        { heading }
      </AccordionSummary>
      <div className={expanded ? classes.detailExpanded : classes.detailCollapsed}>
        <AccordionDetails>
          {body}
        </AccordionDetails>
      </div>
      
    </Accordion>

    
  );
}