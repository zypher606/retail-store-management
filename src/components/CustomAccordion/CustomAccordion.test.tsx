import React from 'react';
import ReactDOM, { unmountComponentAtNode } from 'react-dom';
import { isTSAnyKeyword } from '@babel/types';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { act } from "react-dom/test-utils";
import '@testing-library/jest-dom';
import { v4 as uuidv4 } from 'uuid';
import { CustomAccordion } from './CustomAccordion';

afterEach(() => {
  // cleanup on exiting
  cleanup();
});

describe("CustomAccordion component tests", () => {
  it("rendered successful", () => {
    const div = document.createElement("div");
    render(
      <CustomAccordion 
        openDefault={true}
        fixed={false}
        heading="my heading for accordion"
        body="my body for accordion"
      />
    );
  });
  
  it('matches snapshot', () => {
    const tree = renderer.create(
      <CustomAccordion 
        openDefault={true}
        fixed={false}
        heading="my heading for accordion"
        body="my body for accordion"
      />
    ).toJSON();
    expect(tree).toMatchSnapshot()
  });
  
});