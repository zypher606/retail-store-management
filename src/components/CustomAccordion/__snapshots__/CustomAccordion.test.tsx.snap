// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomAccordion component tests matches snapshot 1`] = `
<div
  className="MuiPaper-root MuiAccordion-root WithStyles(ForwardRef(Accordion))-root-11 Mui-expanded WithStyles(ForwardRef(Accordion))-expanded-12 MuiPaper-elevation1"
>
  <div
    aria-disabled={false}
    aria-expanded={true}
    className="MuiButtonBase-root MuiAccordionSummary-root WithStyles(ForwardRef(AccordionSummary))-root-13 is-expanded Mui-expanded WithStyles(ForwardRef(AccordionSummary))-expanded-15"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiAccordionSummary-content WithStyles(ForwardRef(AccordionSummary))-content-14 Mui-expanded WithStyles(ForwardRef(AccordionSummary))-expanded-15"
    >
      my heading for accordion
    </div>
  </div>
  <div
    className="MuiCollapse-container MuiCollapse-entered"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner"
      >
        <div
          role="region"
        >
          <div
            className="makeStyles-detailExpanded-9"
          >
            <div
              className="MuiAccordionDetails-root WithStyles(ForwardRef(AccordionDetails))-root-16"
            >
              my body for accordion
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
